### PR TITLE
Buyers DAO DAOImpl

### PR DESCRIPTION
**Pull request by kanmaytacker**
Classes Covered
-Buyers
-Country
-State
-City
-Locality
-Apartment

### CODE REVIEW

File: src/main/java/com/FoodyBuddy/Dao/ApartmentDAO.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.FoodyBuddy.DAO;
+
+import java.util.List;
+import com.FoodyBuddy.Model.Apartment;
+
+public interface ApartmentDAO {
+	public List<Apartment> getAllApartments();
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/ApartmentDAO.java]: Don't over-explicit the names in all the DAO and models

instead if  getAllAppartmenets() make  getAll()

File: src/main/java/com/FoodyBuddy/Model/State.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	public Country getCountry(){
+		return country;
+	}
+	
+	public void setCountry(Country country){
+		this.country = country;
+	}
+	@Override
+	public String toString() {
+		return "State [id=" + id + ", name=" + name + ", country=" + country
```
Comment (junzy) [inline src/main/java/com/FoodyBuddy/Model/State.java:60]: not formatted properly

File: src/main/java/com/FoodyBuddy/Model/Locality.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		return this.city;
+	}
+	
+	public void setCity(City city){
+		this.city = city;
+	}
+
+	@Override
+	public String toString() {
+		return "Locality [id=" + id + ", name=" + name + ", pincode=" + pincode
```
Comment (junzy) [inline src/main/java/com/FoodyBuddy/Model/Locality.java:65]: fix indentation

File: src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	@Override
+	public void insertApartment(Apartment apartment) {
+		session.persist(apartment);
+		logger.info("Apartment saved successfully, Apartment Details = " + apartment);
+	}
+
+	@Override
+	public void updateApartment(Apartment apartment) {
+
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java:32]: Inconsistent spacing . Extra line here

File: src/main/java/com/FoodyBuddy/Dao/Impl/CityDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+	@Override
+	public void insertCity(City city) {
+		session.persist(city);
+		logger.info("City saved successfully, City Details = " + city);
+	}
+
+	@Override
+	public void updateCity(City city) {
+
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/CityDAOImpl.java]: extra line

File: src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	public Apartment getApartment(int id) {		
+		Apartment apartment = (Apartment) session.load(Apartment.class, new Integer(id));
+		logger.info("Apartment loaded successfully, Apartment details = " + apartment);
+		return apartment;
+	}
+
+	@Override
+	public void deleteApartment(int id) {
+		Apartment apartment = (Apartment) session.load(Apartment.class, new Integer(id));
+		if(null != apartment){
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java:57]: make it (apprtment !=null)

File: src/main/java/com/FoodyBuddy/Dao/Impl/CityDAOImpl.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	private static final Logger logger = LoggerFactory.getLogger(CityDAOImpl.class);
+
+	private Session session = null;
+	
+	public CityDAOImpl(Session session){
+		this.session = session;
+	}
+
+	@Override
+	public void insertCity(City city) {
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/CityDAOImpl.java:25]: Make the code consistent .  Passig the id sometimes and and object some times

File: src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+	public void updateApartment(Apartment apartment) {
+
+		session.update(apartment);
+		logger.info("Apartment updated successfully, Apartment Details = " + apartment );
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<Apartment> getAllApartments() {
+		List<Apartment> ApartmentsList = session.createQuery("from Apartment").list();
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java:40]: variable name should start with a lower case letter.

File: src/main/java/com/FoodyBuddy/Dao/Impl/CountryDAOImpl.java [hunk 9]
```diff
@@ -1,55 +1,63 @@
...
+		logger.info("Country updated successfully, Country Details = " + country );
+	}
+
 	@SuppressWarnings("unchecked")
-	public List<Country> list() {
-		Session session = this.sessionFactory.openSession();
-		List<Country> CountryList = session.createQuery("from Country").list();
-		session.close();
-		return CountryList;
+	@Override
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Dao/Impl/CountryDAOImpl.java:38]: remove all @Override tags as we are implementing an interface and not extending class.

File: src/main/java/com/FoodyBuddy/Dao/Impl/StateDAOImpl.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+	@Override
+	public void insertState(State state) {
+		session.persist(state);
+		logger.info("State saved successfully, State Details = " + state);
+	}
+
+	@Override
+	public void updateState(State state) {
+
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Dao/Impl/StateDAOImpl.java:32]: Remove empty line.

File: src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+	@Override
+	public Apartment getApartment(int id) {		
+		Apartment apartment = (Apartment) session.load(Apartment.class, new Integer(id));
+		logger.info("Apartment loaded successfully, Apartment details = " + apartment);
+		return apartment;
+	}
+
+	@Override
+	public void deleteApartment(int id) {
+		Apartment apartment = (Apartment) session.load(Apartment.class, new Integer(id));
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/ApartmentDAOImpl.java:56]: Just pass an object or Make just a single  session operation  here

File: src/main/java/com/FoodyBuddy/Model/State.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+	
+	@Column(name = "country_id")
+	@ManyToOne
+	@JoinColumn(name = "id")
+	private Country country;
+	
+	public State(String name, Country country){
+		this.name = name;
+		this.country = country;
+
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Model/State.java:33]: Remove empty line

File: src/main/java/com/FoodyBuddy/Dao/ApartmentDAO.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+	public Apartment getApartment(int id);
+	public void insertApartment(Apartment apartment);
```
Comment (junzy) [inline src/main/java/com/FoodyBuddy/Dao/ApartmentDAO.java:9]: Change method names to insert, update, delete

**Pull request closed without merging**